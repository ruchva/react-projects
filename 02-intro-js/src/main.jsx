import heroes, { getHeroById } from './data/heroes.js';

//Entendemos las promesas
const promesa = new Promise( (resolve, reject) => {
    setTimeout(() => {
        //resolve('Hola Mundo');
        //reject('Error en Hola Mundo');
        //const mi_heroe = heroes.find( (heroe) => heroe.id === 2); 
        const mi_heroe = getHeroById(2); 
        if (mi_heroe) {
            resolve(mi_heroe);
        } else {
            reject('No se pudo encontrar el héroe');
        }
    }, 1000);
});

promesa.then( (heroe) => {
    console.log('mi heroe: ', heroe);
}
).catch( (err) => {
    console.warn(err);
});



const getHeroByIdAsync = (id) => {
    return new Promise( (resolve, reject) => {
        setTimeout(() => {
            const mi_heroe = getHeroById(id); 
            if (mi_heroe) {
                resolve(mi_heroe);
            } else {
                reject('No se pudo encontrar el héroe');
            }
        }, 3000);
    });
}
getHeroByIdAsync(1)
    .then( console.log )
    .catch( console.warn );


// ejemplo de async await
const getHeroByIdAsync2 = async (id) => {
    try {
        const mi_heroe = await getHeroByIdAsync(id);
        console.log(mi_heroe);
    } catch (error) {
        console.warn(error);
    }
}
getHeroByIdAsync2(1);
getHeroByIdAsync2(10)
    .then( console.log )
    .catch( console.warn );   



// ejemplo fetch api
const apiKey = '7a0c1f2b3e8d4f5a9c6e0b8f1c2e0b8f';
const url = `https://api.themoviedb.org/3/movie/popular?api_key=${apiKey}&language=en-US&page=1`;
fetch(url)
    .then(response => response.json())
    .then(data => {
        console.log(data);
    })
    .catch(error => {
        console.error('Error:', error);
    });


// ejemplo de async await con fetch
const getPopularMovies = async () => {
    try {
        const response = await fetch(url);
        const data = await response.json();
        console.log(data);
    } catch (error) {
        console.error('Error:', error);
    }
}
getPopularMovies();

// ejemplo de async await con fetch y destructuring
const getPopularMovies2 = async () => {
    try {
        const response = await fetch(url);
        const { results } = await response.json();
        console.log(results);
    } catch (error) {
        console.error('Error:', error);
    }
}
getPopularMovies2();

// ejemplo de async await con fetch y destructuring y map
const getPopularMovies3 = async () => {
    try {
        const response = await fetch(url);
        const { results } = await response.json();
        const movies = results.map(movie => ({
            title: movie.title,
            release_date: movie.release_date,
            overview: movie.overview
        }));
        console.log(movies);
    } catch (error) {
        console.error('Error:', error);
    }
}
getPopularMovies3();

// ejemplo de async await con fetch y destructuring y map y filter
const getPopularMovies4 = async () => {
    try {
        const response = await fetch(url);
        const { results } = await response.json();
        const movies = results
            .filter(movie => movie.release_date >= '2020-01-01')
            .map(movie => ({
                title: movie.title,
                release_date: movie.release_date,
                overview: movie.overview
            }));
        console.log(movies);
    } catch (error) {
        console.error('Error:', error);
    }
}
getPopularMovies4();

// ejemplo de async await con fetch y destructuring y map y filter y reduce
const getPopularMovies5 = async () => {
    try {
        const response = await fetch(url);
        const { results } = await response.json();
        const movies = results
            .filter(movie => movie.release_date >= '2020-01-01')
            .map(movie => ({
                title: movie.title,
                release_date: movie.release_date,
                overview: movie.overview
            }))
            .reduce((acc, movie) => acc + movie.overview.length, 0);
        console.log(movies);
    } catch (error) {
        console.error('Error:', error);
    }
}
getPopularMovies5();


// ejemplo de operador ternario
const isTrue = true;
const message = isTrue ? 'Es verdadero' : 'Es falso';
console.log(message);

// ejemplo de operador ternario con destructuring
const person = {
    name: 'Juan',
    age: 30,
    isMarried: false
};
const { name, age, isMarried } = person;
const message2 = isMarried ? `${name} está casado` : `${name} no está casado`;
console.log(message2);



