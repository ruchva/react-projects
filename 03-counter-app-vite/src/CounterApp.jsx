import { useState} from "react";
import PropTypes from "prop-types";

export const CounterApp =({ value }) => { 
    const titulo = 'counter App!';

    //** Hook */
    const [ counter, setCounter ] = useState( value );
    const handleAdd = (event) => {
        setCounter( counter + 1 );
    }
    const handleMin = (event) => {
        setCounter( counter - 1 );
    }
    const handleReset = (event) => {
        setCounter( value );
    }
    //** */

    //** Event Handlers */
    function handleClick(event) {
        console.log(event);
    }
    const handleClick2 = (event) => {
        console.log(event);
    }
    
    return (
        <> 
        <h1> { titulo } </h1>
        <h2> { counter } </h2>                   
        <button onClick={ handleAdd } name="+1"> +1 </button>
        <button onClick={ handleReset } name="reset"> Reset </button>
        <button onClick={ handleMin } name="-1"> -1 </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}