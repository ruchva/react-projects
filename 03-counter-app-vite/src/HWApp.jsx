export const HWApp = () => {
  return (
    <div>
      <h1>Hello World App</h1>
      <p>This is the First component.</p>
    </div>
  );
}