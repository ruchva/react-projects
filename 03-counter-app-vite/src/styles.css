html, body {
    height: 100%;
    background-color: #212324;
    color: white;
    font-family: 'Courier New', Courier, monospace;
    font-size: 16px;
    line-height: 1.5;
    margin: 0;
    padding: 70px;
}

button {
    background-color: #007bff;
    border: none;
    color: white;
    padding: 10px 20px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
}