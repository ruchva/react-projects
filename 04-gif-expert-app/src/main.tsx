import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from './GifExpertApp';
import './styles/styles.module.css';


ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
);
