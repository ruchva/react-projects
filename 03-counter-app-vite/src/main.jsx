import React from "react";
import ReactDOM from "react-dom/client";
import { HWApp } from "./HWApp.jsx";
import { FirstApp } from "./FirstApp.jsx";
import { CounterApp } from "./CounterApp.jsx";
import './styles.css';

ReactDOM.createRoot( document.getElementById('root') ).render(
  <React.StrictMode>
    {/* <HWApp /> */}
    <FirstApp title="Hola, soy Goku"/>
    <CounterApp value={ 100 }/>
  </React.StrictMode>
);
