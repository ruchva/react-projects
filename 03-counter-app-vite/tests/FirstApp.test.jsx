import { render, screen } from '@testing-library/react'; 
import { FirstApp } from '../src/FirstApp';

describe('FirstApp', () => {
    
    const title = 'Hola, soy Goku';
    test('should match snapshot', () => {
        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();
    });
    
    test('should show the title in an h1 tag', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain('Hola, soy Goku');
    });
    
    test('should show the subtitle in a p tag', () => {
        render( <FirstApp title="Hola, soy Goku" subtitle="Soy un subtitulo" /> );
        expect( screen.getByText('Soy un subtitulo') ).toBeTruthy();
    });
    
});