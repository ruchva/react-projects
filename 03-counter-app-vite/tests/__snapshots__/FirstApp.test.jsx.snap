// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp should match snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p />
  <p />
</div>
`;
